src/Microsoft.DotNet.Darc/DarcLib/Models/VirtualMonoRepo/ManifestRecord.cs

```diff
@@ -27,19 +28,27 @@ public string GetPublicUrl()
     {
         var url = RemoteUri;
 
-        if (url.EndsWith(".git"))
+        switch (GitRepoUrlParser.ParseTypeFromUri(url))
         {
-            url = url[..^4];
+            case GitRepoType.GitHub:
+                if (url.EndsWith(".git"))
+                {
+                    url = url[..^4];
```

Comment:
Strong recommendation to not use any pre-computed constants, introduce `const string GIT_EXTENSION = ".git";` and replace the `^4` with "^(GIT_EXTENSION.Length)" to avoid off-by-one errors.

Reply:
Done